swarm/storage/dbstore.go

```diff
@@ -804,7 +759,7 @@ func (s *DbStore) SyncIterator(since uint64, until uint64, po uint8, f func(Key,
 	defer it.Release()
 	it.Seek(sincekey)
 
-	for it.Next() {
+	for it.Valid() {
```

Comment:
why put this back? i thought we said this was obsolete

Reply:
Test TestIterator fails if this is used https://github.com/ethersphere/go-ethereum/pull/216/commits/31124f65bf7ef2bb921f2168b6699a199c0c0d1e. It does not break other tests or benchmarks.